packages/confirm-dialog/src/vaadin-confirm-dialog-mixin.js

```diff
@@ -288,51 +291,46 @@ export const ConfirmDialogMixin = (superClass) =>
         },
       });
       this.addController(this._confirmController);
+    }
 
-      this._overlayElement = this.$.overlay;
+    updated(props) {
+      super.updated(props);
+
+      if (props.has('header')) {
+        this.ariaLabel = this.header || 'confirmation';
+      }
     }
 
     /** @protected */
     __onDialogOpened() {
-      const overlay = this._overlayElement;
-
-      // Teleport slotted nodes to the overlay element.
-      this.__slottedNodes.forEach((node) => {
-        overlay.appendChild(node);
-      });
-
-      const confirmButton = overlay.querySelector('[slot="confirm-button"]');
+      const confirmButton = this.querySelector('[slot="confirm-button"]');
```

Comment:
We could also this `this._confirmButton` here since the reference to it is stored by the component.